source/base/bounding_box.cc

```diff
@@ -0,0 +1,43 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by Giovanni Alzetta
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/bounding_box.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+template <int spacedim, typename Number>
+bool BoundingBox<spacedim,Number>::point_inside (const Point<spacedim, Number> &p) const
+{
+  for (unsigned int i=0; i < spacedim; ++i)
+    {
+      //Bottom left-top right convention: the point is outside if it's smaller than the
+      //first or bigger than the second boundary point
+      if ( p[i] < this->boundary_points.first[i] || this->boundary_points.second[i] < p[i])
```

Comment:
`<=` and `<=`

Reply:
also extend your first test to check that `0.,0.` is inside `0,0 -> 1,1` or alike.

Reply:
I think it boils down to the question of whether you consider the bounding box to be mathematically open or closed. If you want to think of it as closed (for example because you want to allow bounding boxes that contain only a single point), then using `<` here is correct.

Reply:
correct. This should be mentioned in documentation. From practical purposes i think we want it to be closed (i.e. contain all its limit points).

Reply:
Yes, I need them to be closed, that's why I'm using <= (and in future a tolerance could be considered, effectively enlarging the box by 1e-10 or something...).

As for the test currently the first check it does is checking that boundary points are inside (i.e. 0.0. for 0,0-> 1,1); but I think it's a good idea to create a simple "unitary" box and check if all the vertices are included.